src/cli/simple/command.rs

```diff
@@ -342,6 +343,76 @@ pub async fn deploy_upgrade(client: Client) -> anyhow::Result<()> {
     .await
 }
 
+/// Updates the bytecode for user's router contracts.
+pub async fn factory_upgrade(client: Client, path: String) -> anyhow::Result<()> {
+    let code = std::fs::read(path)?;
+
+    ContractCall {
+        method: "factory_upgrade",
+        success_message: "The bytecode of user's router contract has been upgraded successfully",
+        error_message: "Error while upgrading the bytecode of user's router contract",
+    }
+    .proceed(client, code)
+    .await
+}
+
+/// Updates the bytecode version for the given account
+pub async fn factory_upgrade_address_version(
```

Comment:
This function can only be called as a callback from the Engine itself, so I don't think it should be supported in the CLI.

Reply:
Thank you, @birchmd for the notice 👍🏻